From de4640579c5318ba08ec57531ef87d63f5574557 Mon Sep 17 00:00:00 2001
From: Chris Kay <chris.kay@arm.com>
Date: Thu, 21 Aug 2025 14:16:25 +0100
Subject: [PATCH] fix(arm): re-enable console by default in BL31

In c997a8d, the common `arm_bl31_early_platform_setup` function
underwent a consolidation of two large preprocessor branches that were
conditional on whether or not Transfer List support is enabled.

This function would initialise the console via `arm_console_boot_init`
*only* if Transfer List support was disabled. During the consolidation,
this call was removed, such that the behaviour was the same for both
branches.

However, the common `bl31_early_platform_setup2` implementation was not
updated to reflect this change, and so platforms that a) relied on this
common implementation and b) did not enable Transfer List support no
longer initialise the console in BL31.

This change ensures that the common implementation correctly initialises
the console during early BL31 boot.

Change-Id: I332af3932ac70382fbf7a5434c0008807f38f86c
Signed-off-by: Chris Kay <chris.kay@arm.com>
Upstream-Status: Backport [de4640579c5318ba08ec57531ef87d63f5574557]
---
 plat/arm/common/arm_bl31_setup.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/arm/common/arm_bl31_setup.c b/plat/arm/common/arm_bl31_setup.c
index 8ac984fd1..e7dc70bdb 100644
--- a/plat/arm/common/arm_bl31_setup.c
+++ b/plat/arm/common/arm_bl31_setup.c
@@ -343,6 +343,9 @@ void __init arm_bl31_early_platform_setup(u_register_t arg0, u_register_t arg1,
 void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 		u_register_t arg2, u_register_t arg3)
 {
+	/* Initialize the console to provide early debug support */
+	arm_console_boot_init();
+
 	arm_bl31_early_platform_setup(arg0, arg1, arg2, arg3);
 
 	/*
-- 
2.43.0

